examples/step-74/step-74.cc

```diff
@@ -0,0 +1,1060 @@
+        {
+          const double diff = (g[point] - sol_u[point]);
+          difference_norm_square += diff * diff * JxW[point];
+        }
+      copy_data.value += penalty * difference_norm_square;
+    };
+
+    // Assemble interior face terms $\sum_{f\in \partial K}\lbrace \sigma
+    // \left\| [u_h]  \right\|_f^2   +  h_f \left\|  [\nu \nabla u_h \cdot
+    // \mathbf n ] \right\|_f^2 \rbrace$.
+    auto face_worker = [&](const Iterator &    cell,
+                           const unsigned int &f,
+                           const unsigned int &sf,
+                           const Iterator &    ncell,
+                           const unsigned int &nf,
+                           const unsigned int &nsf,
+                           ScratchData &       scratch_data,
+                           CopyData &          copy_data) {
+      const FEInterfaceValues<dim> &fe_iv =
+        scratch_data.reinit(cell, f, sf, ncell, nf, nsf);
+
+      copy_data.face_data.emplace_back();
+      CopyDataFace &copy_data_face = copy_data.face_data.back();
+
+      copy_data_face.cell_indices[0] = cell->active_cell_index();
+      copy_data_face.cell_indices[1] = ncell->active_cell_index();
+
+      const std::vector<double> &        JxW     = fe_iv.get_JxW_values();
+      const std::vector<Tensor<1, dim>> &normals = fe_iv.get_normal_vectors();
+
+      const auto &       q_points   = fe_iv.get_quadrature_points();
+      const unsigned int n_q_points = q_points.size();
+
+      std::vector<double> jump(n_q_points);
+      get_function_jump(fe_iv, solution, jump);
+
+      std::vector<Tensor<1, dim>> grad_jump(n_q_points);
+      get_function_gradient_jump(fe_iv, solution, grad_jump);
+
+      const double h = cell->face(f)->diameter();
+
+      const double extent1 =
+        cell->extent_in_direction(GeometryInfo<dim>::unit_normal_direction[f]);
+      const double extent2 = ncell->extent_in_direction(
+        GeometryInfo<dim>::unit_normal_direction[nf]);
+      const double penalty = compute_penalty(fe.get_degree(), extent1, extent2);
+
+      double flux_jump_square = 0;
+      double u_jump_square    = 0;
+      for (unsigned int point = 0; point < n_q_points; ++point)
+        {
+          u_jump_square += jump[point] * jump[point] * JxW[point];
+          const double flux_jump = grad_jump[point] * normals[point];
+          flux_jump_square +=
+            diffusion_coefficient * flux_jump * flux_jump * JxW[point];
+        }
+      copy_data_face.values[0] =
+        0.5 * h * (flux_jump_square + penalty * u_jump_square);
+      copy_data_face.values[1] = copy_data_face.values[0];
+    };
+
+    auto copier = [&](const CopyData &copy_data) {
+      if (copy_data.cell_index != numbers::invalid_unsigned_int)
+        estimated_error_square_per_cell[copy_data.cell_index] +=
+          copy_data.value;
+      for (auto &cdf : copy_data.face_data)
+        for (unsigned int j = 0; j < 2; ++j)
```

Comment:
I really don't like this way to address interior and exterior faces. `0` := interior, `1`:=exterior. Any suggestions for a better interface?